pkg/component/etcd/serviceaccount/serviceaccount.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serviceaccount
+
+import (
+	"context"
+
+	"github.com/gardener/gardener/pkg/controllerutils"
+
+	gardenercomponent "github.com/gardener/gardener/pkg/operation/botanist/component"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type component struct {
+	client client.Client
+	values *Values
+}
+
+func (c *component) Deploy(ctx context.Context) error {
+	serviceAccount := c.emptyServiceAccount()
+	_, err := controllerutils.GetAndCreateOrStrategicMergePatch(ctx, c.client, serviceAccount, func() error {
+		serviceAccount.Name = c.values.Name
+		serviceAccount.Namespace = c.values.Namespace
+		serviceAccount.Labels = c.values.Labels
+		serviceAccount.OwnerReferences = c.values.OwnerReferences
+		serviceAccount.AutomountServiceAccountToken = pointer.Bool(!c.values.DisableAutomount)
```

Comment:
I am now wondering why did we have a config value to `DisableAutomount`(default value false) instead of `EnableAutomount` (default value true). Then we have to negate it always to set it.